dace/codegen/targets/sve/codegen.py

```diff
@@ -0,0 +1,588 @@
+        offset_2 = offset_1.subs(sve_sym, sve_sym + sve_range[2])
+
+        # 3. The load stride is the difference between both
+        stride = (offset_2 - offset_1).simplify()
+
+        return stride
+
+    def add_header(self, function_stream: CodeIOStream):
+        if self.has_generated_header:
+            return
+        self.has_generated_header = True
+
+        function_stream.write('#include <arm_sve.h>\n')
+
+        # TODO: Find this automatically at compile time
+        function_stream.write(f'#define {util.REGISTER_BYTE_SIZE} 64\n')
+
+    def __init__(self, frame_codegen: DaCeCodeGenerator, sdfg: dace.SDFG):
+        dace.SCOPEDEFAULT_SCHEDULE[
+            dace.ScheduleType.SVE_Map] = dace.ScheduleType.Sequential
+        dace.SCOPEDEFAULT_STORAGE[
+            dace.ScheduleType.SVE_Map] = dace.StorageType.CPU_Heap
+        self.has_generated_header = False
+
+        """
+        # Hook the generate_header in DaCeCodeGenerator
+        global old_generate_footer, old_generate_header
+        if old_generate_footer is None:
+            old_generate_header = DaCeCodeGenerator.generate_header
+            old_generate_footer = DaCeCodeGenerator.generate_footer
+        DaCeCodeGenerator.generate_header = generate_header_hook
+        DaCeCodeGenerator.generate_footer = generate_footer_hook
+        """
+
+        self.frame = frame_codegen
+        self.dispatcher = frame_codegen._dispatcher
+        self.dispatcher.register_map_dispatcher(
+            dace.ScheduleType.SVE_Map, self)
+        self.dispatcher.register_node_dispatcher(
+            self, lambda state, sdfg, node: is_in_scope(
+                state, sdfg, node, [dace.ScheduleType.SVE_Map]))
+        #self.dispatcher.register_state_dispatcher(self, lambda sdfg, state: contains_any_sve(sdfg))
+
+        self.cpu_codegen = self.dispatcher.get_generic_node_dispatcher()
+        self.state_gen = self.dispatcher.get_generic_state_dispatcher()
+
+        for src_storage, dst_storage in itertools.product(
+                dtypes.StorageType, dtypes.StorageType):
+            self.dispatcher.register_copy_dispatcher(src_storage, dst_storage,
+                                                     dace.ScheduleType.SVE_Map,
+                                                     self)
+
+    def create_empty_definition(self,
+                                conn: dace.typeclass,
+                                edge: gr.MultiConnectorEdge[mm.Memlet],
+                                callsite_stream: CodeIOStream,
+                                output: bool = False,
+                                is_interconnect: bool = False):
+        """ Creates a simple variable definition `type name;`, which works for both vectors and regular data types. """
+
+        var_name = None
+        var_type = None
+        var_ctype = None
+
+        if output:
+            var_name = edge.dst_conn
+        else:
+            var_name = edge.src_conn
+
+        if is_interconnect:
```

Comment:
I would prefer if you didn't define new names where current ones exist. These are called code->code edges (because they are directly between two `CodeNode`s)